packages/ai/src/ai/account/store_providers/azure.py

```
@@ -307,6 +307,119 @@ async def list_files(self, prefix: str = '') -> list:
+    async def write_chunk(self, filename: str, context, data: bytes) -> int:
+        """Buffer data and stage blocks when buffer reaches 4 MB."""
+        try:
+            context['buffer'].extend(data)
+            written = len(data)
+
+            while len(context['buffer']) >= self._AZURE_BLOCK_SIZE:
+                chunk = bytes(context['buffer'][: self._AZURE_BLOCK_SIZE])
+                del context['buffer'][: self._AZURE_BLOCK_SIZE]
+                self._stage_block(context, chunk)
+
+            return written
```

Ruff (0.15.7)

[warning] Consider moving this statement to an `else` block

(TRY300)